src/Character.java

```diff
@@ -0,0 +1,164 @@
+
+public abstract class Character {
+
+	// Attributes
+	private static int lastId = 0;
+	private int id;
+	private byte level;
+	private static byte maxLevel = 50;
+	private static short baseAttack = 5;
+	private static short baseDefense = 2;
+	private static short baseHp = 50;
+	private String name;
+	protected static final String errorMessage = "Error: Invalid value";
+	
+	// Constructor
+	public Character(byte level, String name) {
+		this.setId();
+		this.setLevel(level);
+		this.setName(name);
+	}
+
+	// Methods
+	/**
+	 * Abstract method to calculate the attack (damage) of a character
+	 * 
+	 * This method will be defined and implemented by the child classes
+	 * */
+	public short calculateAttack() {
+		short random = (short)(Math.random() * (1 - 0)); 
+		return (short) (Character.baseAttack  + ( Character.baseAttack * random ) * this.level);
+	}
+	
+	/**
+	 * Abstract method to calculate the defense of a character
+	 * 
+	 * This method will be defined and implemented by the child classes
+	 * */
+	public short calculateDefense() {
+		return (short) ((Character.baseDefense * this.level) + (this.calculateHp() * 0.05));
+	}
+	
+	/**
+	 * Abstract method to calculate the hp of a character
+	 * 
+	 * This method will be defined and implemented by the child classes
+	 * */
+	public short calculateHp() {
+		return (short) ((Character.baseHp * this.level) - (0.5 * this.calculateAttack()));
+	}
+
+	// Override equals method.
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (obj == null)
+			return false;
+		if (getClass() != obj.getClass())
+			return false;
+		Character other = (Character) obj;
+		if (id != other.id)
+			return false;
+		if (level != other.level)
+			return false;
+		if (name == null) {
+			if (other.name != null)
+				return false;
+		} else if (!name.equals(other.name))
+			return false;
+		return true;
+	}
+
+	// Override toString method.
+	@Override
+	public String toString() {
+		return "Character [getId()=" + getId() + ", getLevel()=" + getLevel() + ", getName()=" + getName() + "]";
```

Comment:
I would modify toString so it doesn't show "getId() ="+ getId(), but "Id = " + getId() instead

Reply:
Done!